### PR TITLE
Commits on AD review of draft-ietf-uta-smtp-tlsrpt-04.txt

### PR DESCRIPTION
**Pull request by prbinu**
Fixes:
1. Replaced SSL with TLS
2. Replaced text/json with application/json
3. Add RFC references
3. Modified 'related technologies' section

### CODE REVIEW

**Comment by prbinu** [comment]
@danmarg @abrotman @mrisher 
One of you please review and merge

File: reporting.md [hunk 1]
```diff
@@ -114,11 +114,9 @@ We also define the following terms for further use in this document:
 
 * This document is intended as a companion to the specification for SMTP MTA
     Strict Transport Security (MTA-STS, TODO: Add ref).
-* The Public Key Pinning Extension for HTTP [@!RFC7469] contains a JSON-based
-    definition for reporting individual pin validation failures.
-* The Domain-based Message Authentication, Reporting, and Conformance (DMARC)
```
Comment (danmarg) [inline reporting.md]: Why remove these references (to HPKP and DMARC reporting)?

File: reporting.md [hunk 2]
```diff
@@ -137,7 +135,7 @@ Policies consist of the following directives:
   * In the case of `https`, reports should be submitted via POST
            ([@!RFC2818]) to the specified URI.
   * In the case of `mailto`, reports should be submitted to the specified
-           email address. When sending failure reports via SMTP, sending MTAs
+           email address ([@!RFC6068]). When sending failure reports via SMTP, sending MTAs
```
Comment (danmarg) [inline reporting.md]: Nit: 80 col line wrap? ;)

File: reporting.md [hunk 3]
```diff
@@ -499,10 +497,9 @@ Figure: JSON Report Format
     Presently, the only three valid choices are `tlsa`, `sts`, and the literal
     string `no-policy-found`. It is provided as a string.
 * `policy-string`: The string serialization of the policy, whether TLSA record
-    or MTA-STS policy. Any linefeeds from the original policy MUST be replaced with
-    [SP]. TODO: Help with specifics.
+    or MTA-STS policy. Any linefeeds from the original policy MUST be replaced with space.
```
Comment (danmarg) [inline reporting.md]: For TLSA records, is it worth adding a reference to section 2.3 here (https://tools.ietf.org/html/rfc6698#section-2.3) for how to present it? 

I also do not understand why linefeeds are not valid; JSON strings can handle newline characters in strings (via "\n").

File: reporting.md [hunk 4]
```diff
@@ -499,10 +497,9 @@ Figure: JSON Report Format
...
 * `domain`: The Policy Domain upon which the policy was applied. For messages
-    sent to `user@example.com` this field would contain `example.com`. It is
+    sent to `user@example.com` this field would contain `example.com` ([@!RFC5890]). It is
```
Comment (danmarg) [inline reporting.md]: Is a reference to 5890 actually helpful here? This is not specific to IDN.

File: reporting.md (hunk 1 above)
Comment (prbinu) [inline reporting.md]: Unlike DNSSEC & DANE, these are not directly related to MTA-STS. IMO these are 'good-to-have' but not required to be mentioned.

Comment (danmarg) [inline reporting.md]: Hmm, OK. I think it would be nice to reference them somewhere just so people can get a sense of similar "reporting" technologies. But I agree they aren't critical here.

File: reporting.md [hunk 5]
```diff
@@ -499,11 +497,11 @@ Figure: JSON Report Format
     Presently, the only three valid choices are `tlsa`, `sts`, and the literal
     string `no-policy-found`. It is provided as a string.
 * `policy-string`: The string serialization of the policy, whether TLSA record
-    or MTA-STS policy. Any linefeeds from the original policy MUST be replaced with
-    [SP]. TODO: Help with specifics.
-* `domain`: The Policy Domain upon which the policy was applied. For messages
-    sent to `user@example.com` this field would contain `example.com`. It is
-    provided as a string.
+    ([@!RFC6698] section 2.3) or MTA-STS policy. Any linefeeds from the original
+    policy MUST be replaced with space.
```
Comment (danmarg) [inline reporting.md]: But I still don't understand the requirement to replace linefeeds. JSON represents linefeeds inside of strings just fine, right? So why mangle the original policies at all?

File: reporting.md [hunk 6]
```diff
@@ -499,11 +497,11 @@ Figure: JSON Report Format
...
+* `domain`: The Policy Domain is the fully qualified DNS domain name ([@!RFC1035])
+    upon which the policy was applied ([@!RFC5890] for international domains). For
```
Comment (danmarg) [inline reporting.md]: I still don't read this properly. RFC5890 and the "for international domains" reference implies that the "domain" is only the Policy Domain for IDN domains, but of course it's the Policy Domain in all cases. I am not sure an IDN reference is even needed here; I would say something like

> The Policy Domain is the domain against which the MTA-STS or DANE policy is defined

as we say in Terminology. I'm also not sure the 1035 reference really adds much. (If this was a response to comments from the AD review, forgive me; I have not yet read it!)

We could (if we really wanted to) specify whether A form or U form should be used, but I don't see why, since these are mostly for human consumption.

File: reporting.md (hunk 5 above)
Comment (danmarg) [inline reporting.md]: Sure. See the string representation section here: https://tools.ietf.org/html/rfc7159#section-7. 

My point was that in JSON we don't have to replace the line feed character (since that means mangling the original policy); we merely need to escape it (e.g. "/n" and "/r"). I am not sure we need to specify this very carefully since "string representation of the policy" in "JSON" more or less implies as much, no?

File: reporting.md (hunk 6 above)
Comment (danmarg) [inline reporting.md]: *shrug* 

That's a very careful specification. We can keep it, but I'm not very clear on the wording as you have it. (Sorry!) But since we define "Policy Domain" in the "terminology" section, I would not neecssarily bother. I also do not believe we need to specify whether to use A form or U form; it matters for TLSA because they need to specify how to match the domain with the cert, but here it's just for reporting and recipients can presumably figure it out.

File: reporting.md (hunk 5 above)
Comment (prbinu) [inline reporting.md]: Updated based on your comments.

File: reporting.md (hunk 6 above)
Comment (prbinu) [inline reporting.md]: Updated based on your comments.

**Comment by danmarg** [comment]
LGTM. Thanks for your patience.
